supervisor/process.py

```diff
@@ -209,6 +211,9 @@ def spawn(self):
         self._assertInState(ProcessStates.EXITED, ProcessStates.FATAL,
                             ProcessStates.BACKOFF, ProcessStates.STOPPED)
 
+        if delaystart:
+            time.sleep(self.config.startdelay)
```

Comment:
I could not agree more. Unfortunately the extent of modifications required to implement this in an asynchronous manner will never get accepted. With this ugly hack I've just made a poorly implemented feature which desperate people like myself may use to quickly prevent machine crashes when starting all supervised programs (because they all get started at the exact same time).

As for your example: that will only delay my problem 5 seconds.
Because I need each program to get started N seconds after the previous one was started. Eg: 3 programs with a 5 second start delay: prog. 1 at 12:00:00, prog. 2 at 12:00:05 and prog 3 at 12:00:10 - this will distribute the startup peak load.